//
// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package emulator.tests.common;

import "tests/common/test.proto";

message ImportingParent {
  optional string field = 1;
  optional Simple imported = 2;
}

message ImportingAndNestingEnumContainer {
  enum TestEnum {
    TEST_ENUM_UNSPECIFIED = 0;
    TEST_ENUM_ONE = 1;
    TEST_ENUM_TWO = 2;
    TEST_ENUM_THREE = 3;
    TEST_ENUM_SIX = 6;
  }
  optional TestEnum local_enum = 1;
  optional .emulator.tests.common.TestEnum imported_global_enum = 2;
  optional .emulator.tests.common.EnumContainer.TestEnum
      imported_namespaced_enum = 3;
}

message ImportingAndParentingLevelTwo {
  optional string field = 1;
  repeated Parent from_another_file = 2;
  repeated ImportingParent importing_parent_but_local_container = 3;
  optional ImportingAndNestingEnumContainer nested_enum_container = 4;
  optional .emulator.tests.common.TestEnum imported_enum = 5;
}
